src/core/server/http/http_config.ts

```diff
@@ -38,7 +39,17 @@ export const config = {
           validate: match(validBasePathRegex, "must start with a slash, don't end with one"),
         })
       ),
-      defaultRoute: schema.maybe(schema.string()),
+      csp: schema.object(
+        {
+          directives: schema.string(),
+          rules: schema.arrayOf(schema.string()),
+          strict: schema.boolean(),
+          warnLegacyBrowsers: schema.boolean(),
+        },
+        {
+          defaultValue: DEFAULT_CSP_OPTIONS,
```

Comment:
https://github.com/elastic/kibana/blob/bd3857a79befd9baa88e0ca76f6b1594f8716b88/src/legacy/server/config/schema.js#L55-L59

In legacy, `csp` is not included in `server`. I think it makes sense to move it to server config, however we need to be careful, as in current PR state, the properties are dissociated. Probably gonna needs to add deprecations for this once https://github.com/elastic/kibana/pull/52251 lands (if we really want to move `csp` under `server`)